framework/src/modules/chain/synchronizer/utils.js

```diff
@@ -42,6 +45,44 @@ const restoreBlocks = async (blocksModule, processorModule, tx) => {
 	return true;
 };
 
+/**
+ * Allows to restore blocks if there are blocks left upon startup in temp_block table (e.g. node crashed)
+ * Depends upon fork choice rule if blocks will be applied
+ *
+ * 1. Gets all blocks from temp_block table
+ * 2. Sort blocks according to height as we want to lowest height (the next block to be applied)
+ * 3. Uses next block with fork choice rule - if fork status indicates we should switch to different chain
+ * we continue applying blocks using the `restoreBlocks` function.
+ * Otherwise we truncate the temp_block table.
+ *
+ * @param {Object} blocksModule - injection of blocks module object
+ * @param {Object} processorModule - injection of processor module object
+ * @param {Object} storageModule - injection of storage module object
+ * @return {Promise<void>}
+ */
+const restoreBlocksUponStartup = async (
+	blocksModule,
+	processorModule,
+	storageModule,
+) => {
+	// Get all blocks and find lowest height (next one to be applied)
+	const tempBlocks = await storageModule.entities.TempBlock.get();
+	const blockLowestHeight = tempBlocks.reduce((prev, current) =>
+		prev.height < current.height ? prev : current,
+	);
+	const nextTempBlock = parseBlockToJson(blockLowestHeight.fullBlock);
```

Comment:
do we need to do this step?
I think rather than `toJSON`, it should be deserialized to pass to the processor